internal/services/recoveryservices/recovery_services_vault_resource.go

```diff
@@ -324,6 +333,19 @@ func resourceRecoveryServicesVaultCreate(d *pluginsdk.ResourceData, meta interfa
 		return fmt.Errorf("waiting for on update for Recovery Service %s: %+v", id.String(), err)
 	}
 
+	if d.Get("classic_vmware_replication_enabled").(bool) {
+		settingsId := replicationvaultsetting.NewReplicationVaultSettingID(id.SubscriptionId, id.ResourceGroupName, id.VaultName, "default")
+		settingsInput := replicationvaultsetting.VaultSettingCreationInput{
+			Properties: replicationvaultsetting.VaultSettingCreationInputProperties{
+				VMwareToAzureProviderType: utils.String("Vmware"),
+			},
+		}
+		err := settingsClient.CreateThenPoll(ctx, settingsId, settingsInput)
+		if err != nil {
+			return fmt.Errorf("creating %s: %+v", settingsId, err)
+		}
```

Comment:
This can become
```suggestion
		if err := settingsClient.CreateThenPoll(ctx, settingsId, settingsInput); err != nil {
			return fmt.Errorf("creating %s: %+v", settingsId, err)
		}
```